logbook-core/src/main/java/org/zalando/logbook/core/attributes/JwtBaseExtractor.java

```diff
@@ -0,0 +1,71 @@
+package org.zalando.logbook.core.attributes;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+import org.zalando.logbook.HttpHeaders;
+import org.zalando.logbook.HttpRequest;
+import org.zalando.logbook.attributes.AttributeExtractor;
+
+import javax.annotation.Nonnull;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+/**
+ * Extracts a single claim from the JWT bearer token in the request Authorization header.
+ * By default, the subject claim "sub" is extracted, but you can pass an (ordered) list of <code>claimNames</code>
+ * to be scanned. The first claim in <code>claimNames</code> is then returned, or an empty attribute if no matching
+ * claim is found.
+ */
+@API(status = EXPERIMENTAL)
+@Slf4j
+@AllArgsConstructor
+@EqualsAndHashCode
+public abstract class JwtBaseExtractor implements AttributeExtractor {
```

Comment:
Personally I'm not a fan of implementation inheritance and I pretty much avoid it 100% of the time.
You could solve the need to share the claims extraction here by simply having a dedicated `ClaimsExtractor` class that provides that functionality. Any JWT extractor could then just use that as an instance field (composition) instead of a shared base class (inheritance).

Reply:
Fixed in cd098b2.